### ISSUE TITLE
Tracks 2.6.1 Container won't start after it has ran once

### ISSUE THREAD

**Issue by john2exonets** [TracksApp/tracks#2844]
You can start the Tracks 2.6.1 Container just once.....after that, it won't start and a 'docker logs tracks' dump will show this error when you try and start it again:

```
=> Booting Puma
=> Rails 6.0.5.1 application starting in production
=> Run `rails server --help` for more startup options
A server is already running. Check /app/tmp/pids/server.pid.
Exiting

```

**Comment by ZeiP** [comment]
Which installation method and which commands are you using?

**Comment by john2exonets** [comment]
i am following the Installation guide from this repo:  https://github.com/TracksApp/tracks/blob/master/doc/installation.md

**Comment by cosmoneer** [comment]
I'm having the same problem with a Docker container running on UNRAID 6.9.2, using the UNRAID Community Apps feature to install it.  It appears to be using this repository:  https://hub.docker.com/r/tracksapp/tracks

This is the error I get:
=> Booting Puma
=> Rails 6.0.4.6 application starting in production
=> Run `rails server --help` for more startup options
A server is already running. Check /app/tmp/pids/server.pid.

There is an installation note with the container, which reads:
> NOTE: After installing, you must console into the container and run the following command to initialize the database first!
> rake db:reset
> After running that command, you should see the database has tables in it and the app should be usable at that point.

I successfully executed this command and have not had any issues.  A reboot of the host server will restore functionality, but only so long as the container is not stopped.

**Comment by blacktav** [comment]
I am also experiencing this problem.
I am using

- the latest tracksapp on docker (tho the problem existed on a previous image as well)
- an external postgreSQL database (14.6)
- an ArchLinux host (updated @ 28/12/2022)

The container can be created fine but after stopping and attempting to restart, the following is logged:
```
=> Booting Puma
=> Rails 6.0.5.1 application starting in production 
=> Run `rails server --help` for more startup options
A server is already running. Check /app/tmp/pids/server.pid.
```
Obviously I cannot reinitialise my database as @cosmoneer suggests
When the container is first run-up, it logs this...
```
=> Booting Puma
=> Rails 6.0.5.1 application starting in production 
=> Run `rails server --help` for more startup options
Puma starting in single mode...
* Puma version: 6.0.0 (ruby 2.7.7-p221) ("Sunflower")
*  Min threads: 5
*  Max threads: 5
*  Environment: production
*          PID: 9
* Listening on http://0.0.0.0:3000
Use Ctrl-C to stop
```

**Comment by blacktav** [comment]
The container can be reliably rebuilt at anytime using the existing database.
It is just the start/stop cycle do not function

**Comment by nrybowski** [comment]
I had the same issue with the `/app` folder being a volume mounted from the host in the container. Removing the file `/app/tmp/pids/server.pid` before restarting the container solves the issue.

**Comment by LeeThompson** [comment]
Yup this happens to me too.     The container version should probably not check to see if the server is running like this at all.

**Comment by ZeiP** [comment]
It indeed seems that for some reason the server doesn't remove the pid file when exiting. This is likely because of a failed exit; in a local environment this is easy to workaround just by removing the tmp/pids/server.pid file before restarting, but need to figure out why it's not exiting properly.

**Comment by LeeThompson** [comment]
Actually in a container it's not that easy to remove it especially if docker is running on an appliance (NAS, etc).     It may be best to have the container version not check the pid at all.     Currently, when it sees the pid is place the container fatals out making it difficult (unless you're really familiar with docker tools) to remove the pid from the volume since you can't attach a terminal to the container (since it's no longer running).

(I ended up working around this by mounting the pid file on the host file system and writing a bash script to remove it (if present) on system startup.)

Mostly this mechanism just make it harder to start Tracks if there is an abnormal shutdown or if it doesn't clean up the pid.

**Comment by john2exonets** [comment]
Sadly, this is still happening in the latest version (2.7.1) ....two years later.....

**Comment by blacktav** [comment]
You can workaround this by removing the pid file immediately before the container is stopped, for example:
`docker exec <NAME> rm /app/tmp/pids/server.pid`

**Comment by LeeThompson** [comment]
> You can workaround this by removing the pid file immediately before the container is stopped, for example: `docker exec <NAME> rm /app/tmp/pids/server.pid`

That doesn't work very well when the container stops nearly instantly.

**Comment by evmcl** [comment]
I've found adding the arguments `--tmpfs /app/tmp` to my `docker run` command when setting up the container fixes this. That way the `/app/tmp` folder is empty each time the container starts.
